b2share/modules/communities/api.py

```diff
@@ -91,12 +91,21 @@ def get(cls, id=None, name=None, with_deleted=False):
     @classmethod
     # TODO: change this into a search function, not just a list of communities
     # TODO: a query should be given
-    def get_all(cls, start, stop):
+    def get_all(cls, start=None, stop=None):
         """Searches for matching communities."""
-        start = int(start)
-        stop = int(stop)
         from .models import Community as CommunityMeta
-        metadata = CommunityMeta.query.order_by(CommunityMeta.created).limit(stop)[start:]
+        if (start==None and stop==None):
+            metadata = CommunityMeta.query.order_by(CommunityMeta.created)
+        elif (start and stop):
+            try:
+                start = int(start)
```

Comment:
@thijscobben Indeed. This REST API code should also be refactored in order to follow [the same pagination style as in invenio-records-rest](https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L429), i.e. `page` and `size`.